### ISSUE TITLE
Render rounded corners via autoscaling buffer

### ISSUE THREAD

**Issue by Consolatis** [labwc/labwc#593]
Otherwise blurry galore happens:

![image](https://user-images.githubusercontent.com/35009135/196048629-196135fa-c40d-4c99-b9dc-fcf10b7ab241.png)

**Comment by Consolatis** [comment]
I looked into this a while ago and for this to be implemented we have to modify the `scaled_scene_buffer` API a bit.
As of right now the buffer implementation will automatically free the backing `wlr_buffer` when the corresponding scene node is destroyed which is something we really don't want for the SSD images as we keep reusing the backing `wlr_buffer` for all future SSDs.

**Comment by Consolatis** [comment]
@tokyo4j just to confirm, we are now doing that with the scaled_img_buffer, correct? If yes this issue can be closed.

**Comment by tokyo4j** [comment]
No. Icons are rendered with `scaled_img_buffer`, but titlebar corners placed below them can still be blurry.

I actually implemented `scaled_corner_buffer` for that, but I found its border width becomes consistent with other straight borders when fractional scales are used, in a the same manner as #2394. I'll open a draft PR for further discussion later.

**Comment by lszl84** [comment]
Any update on this one? The README proudly mentions HiDPI support, but blurry window decorations really spoil the experience...

**Comment by tokyo4j** [comment]
> Any update on this one? The README proudly mentions HiDPI support, but blurry window decorations really spoil the experience...

Sorry for your inconvenience. I think we can fix this problem with the approach proposed here (#2695), which draws the titlebar as a single auto-scaling buffer. We will definitely need to fix this problem.
